ddtrace/tracer/metrics_test.go

```diff
@@ -0,0 +1,68 @@
+package tracer
+
+import (
+	"sync"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/ext"
+)
+
+func TestReportMetrics(t *testing.T) {
+	trc := &tracer{
+		stopped: make(chan struct{}),
+		config: &config{
+			serviceName: "my-service",
+			hostname:    "my-host",
+			globalTags:  map[string]interface{}{ext.Environment: "my-env"},
+		},
+	}
+
+	var tg testGauger
+	go trc.reportMetrics(&tg, time.Millisecond)
+	time.Sleep(5 * time.Millisecond)
+	close(trc.stopped)
+	assert := assert.New(t)
+	calls := tg.CallNames()
+	tags := tg.Tags()
+	assert.True(len(calls) > 30)
```

Comment:
Correct. The point is to test that some do get reported. I thought it would be tedious to have a long piece of test here which asserts each and every metric.